### ISSUE TITLE
Linux builds?

### ISSUE THREAD

**Issue by dromer** [Kiriki-liszt/JS_Inflator#22]
The Readme says `Linux not compatible.` but it can be built:

![](https://i.imgur.com/K2YlonB.png)

**Comment by Kiriki-liszt** [comment]
Hello!
Thank you for checking it out!
I don't have Linux running laptop anymore, and a bit lazy to make a bootable SSD...
Since I doesn't provide a Linux build, I just said it's not Linux compatible.
We can now safely say 'it's Linux compatible, but you have to build it yourself'
I'll update README!
Have a nice day =)

**Comment by dromer** [comment]
Yeah it does currently require to modify the CMakeLists.txt to point to the vst3sdk, but this is anyway the case for anyone not building on your system.

Maybe you can migrate to a gitmodule for the sdk and use github Actions to do the build for you.
Then have it upload the binaries for the release.

cheers

**Comment by glitterball** [comment]
I have been running the windows version under wine in Linux, however the meters don't work correctly.
A native linux build would be awesome.

**Comment by ghost** [comment]
> The Readme says `Linux not compatible.` but it can be built:
> 
> ![](https://camo.githubusercontent.com/392bed85d7a48543b949baca36125747b4db8aa8a0dedb446fdaf9b560a3fc19/68747470733a2f2f692e696d6775722e636f6d2f4b32596c6f6e422e706e67)

Hi,

How do I build for Ubuntu distribution? Is it possible to do so? A Native Linux Build is planned by owner? (PLEASE...)
cheers

**Comment by dromer** [comment]
> How do I build for Ubuntu distribution?

Modify the `CMakeLists.txt` pointing `vst3sdk_SOURCE_DIR` to your VST3SDK directory.

**Comment by ghost** [comment]
> > How do I build for Ubuntu distribution?
> 
> Modify the `CMakeLists.txt` pointing `vst3sdk_SOURCE_DIR` to your VST3SDK directory.

Oh, thank you so much! Now I try......
cheers

**Comment by ghost** [comment]
Hi,
nothing, I hope to see the birth of the official Linux version (.clap or .vst3) of this and also for Relief_Compressor....

regards

**Comment by dromer** [comment]
Try this fix: https://github.com/Kiriki-liszt/JS_Inflator/pull/47

For me this generates a successful build:

```sh
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DSMTG_ENABLE_VSTGUI_SUPPORT=ON -DSMTG_ENABLE_VST3_PLUGIN_EXAMPLES=OFF -DSMTG_ENABLE_VST3_HOSTING_EXAMPLES=OFF ..
make -j
```

**Comment by ghost** [comment]
> Try this fix: [#47](https://github.com/Kiriki-liszt/JS_Inflator/pull/47)
> 
> For me this generates a successful build:
> 
> mkdir build
> cd build
> cmake -DCMAKE_BUILD_TYPE=Release -DSMTG_ENABLE_VSTGUI_SUPPORT=ON -DSMTG_ENABLE_VST3_PLUGIN_EXAMPLES=OFF -DSMTG_ENABLE_VST3_HOSTING_EXAMPLES=OFF ..
> make -j

Dear Dromer, 
this is not the first time you have helped me. Thank you very much! 
Not being an expert user of Linux, I find it difficult to build this plugin as well. The only build I ran successfully is that of csound and paulxstretch..... Simple git cloning of the repository and just a couple of other commands..... But this is difficult for me. At this point I sincerely hope for the release of an official version from the developer.....
regards

**Comment by esteakacrownie** [comment]
I've tried building it from source on linux (Arch linux kernel 6.18.7), and i'm getting this error

<img width="1290" height="86" alt="Image" src="https://github.com/user-attachments/assets/35858efa-2e2e-4f54-8db6-7fbcb606af3a" />

`VST_SDK/JS_Inflator/source/JSIF_processor.h:316:33: error: declaration of ‘std::vector<yg331::_BS> yg331::JSIF_Processor::Band_Split’ changes meaning of ‘Band_Split’ [-Wchanges-meaning]
  316 |         std::vector<Band_Split> Band_Split;
`
Does anyone know why this is happening, and if not, could anyone who manages to build the linux vst3 successfully share a build ?

**Comment by dromer** [comment]
@esteakacrownie try the branch of my PR, it fixes this issue iirc -> https://github.com/Kiriki-liszt/JS_Inflator/pull/47
